### PR TITLE
chore(ci): CI with Github Actions

### PR DESCRIPTION
**Pull request by strophy**
## Issue being fixed or feature implemented
We experiencing several problems with Travis CI. It is slow and does not allow to restart builds.

## What was done?
Introduce Github Action workflows for running tests and releasing.

## How Has This Been Tested?
With Github Actions

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by strophy** [comment]
I think this is ready for review. It doesn't matter that private keys are exposed in the "Parametrize environment" step because these keys are only useful on the test runner, right?

File: .github/workflows/test_and_release.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+        with:
+          concurrency: 8
+
+      - name: Install NPM dependencies
+        run: npm ci
+
+      - name: Create dotenv file
+        run: cp .env.example .env
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:41]: I guess there is no point to copy .env if we create it down below

File: .github/workflows/test_and_release.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          install-sdk-version: ${{ steps.runtime-vars.outputs.install-sdk-version }}
+
+      - name: Parametrize environment
+        run: |
+          echo "DAPI_SEED=127.0.0.1" > .env
+          echo "FAUCET_PRIVATE_KEY=${{ env.faucet-private-key }}" >> .env
+          echo "NETWORK=regtest" >> .env
+          echo "DPNS_CONTRACT_ID=${{ env.dpns-contract-id }}" >> .env
+
+      - name: Check NPM package lock version is updated
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:72]: I think the efficient order will be:
1. enable NPM cache
2. install NPM deps
3. Run check package
4. run linter
5. docker cache
5. dashevo/gh-action-runtime-vars@master
6. dashevo/gh-action-platform-branches@master
7.  dashevo/gh-action-start-local-network@master
8. Create .env file
9. Run tests
10. show docker

File: .github/workflows/test_and_release.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  pull_request:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+-dev
+
+jobs:
+  test:
+    name: Run wallet-lib tests
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:16]: 15

Comment (shumkov) [inline .github/workflows/test_and_release.yml:16]:
```suggestion
    timeout-minutes: 15
```

File: .github/workflows/test_and_release.yml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  release:
+    types: [published]
+  pull_request:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+-dev
+
+jobs:
+  test:
+    name: Run wallet-lib tests
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:14]:
```suggestion
    name: Run Wallet lib tests
```

File: .github/workflows/test_and_release.yml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      - uses: dashevo/gh-action-start-local-network@master
+        with:
+          actions-cache-url: ${{ steps.platform-branches.outputs.cache-url }}
+          actions-cache-token: ${{ steps.platform-branches.outputs.cache-token }}
+          drive-branch: ${{ steps.runtime-vars.outputs.drive-branch }}
+          dapi-branch: ${{ steps.runtime-vars.outputs.dapi-branch }}
+          dashmate-branch: ${{ steps.platform-branches.outputs.dashmate-branch }}
+          install-sdk-version: ${{ steps.runtime-vars.outputs.install-sdk-version }}
+
+      - name: Parametrize environment
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:68]:
```suggestion
      - name: Create dotenv file
```

File: .github/workflows/test_and_release.yml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      - name: Run ESLinter
+        run: npm run lint
+
+      - name: Cache Docker layers
+        uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+        with:
+          concurrency: 8
+
+      - uses: dashevo/gh-action-runtime-vars@master
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:46]: - name: Parse variables from PR body

File: .github/workflows/test_and_release.yml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        id: runtime-vars
+        if: ${{ contains(github.event.pull_request.body, '/set-var')}}
+        with:
+          pull-request-body: ${{ github.event.pull_request.body }}
+
+      - uses: dashevo/gh-action-platform-branches@master
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:52]: - name: Set compatible branches for Platform components

File: .github/workflows/test_and_release.yml [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+        id: platform-branches
+        with:
+          override-major-version: 0
+          override-dashmate-branch: ${{ steps.runtime-vars.outputs.dashmate-branch }}
+          override-testsuite-branch: ${{ steps.runtime-vars.outputs.testsuite-branch }}
+
+      - uses: dashevo/gh-action-start-local-network@master
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:59]: - name: Start Dash local network

File: .github/workflows/test_and_release.yml [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+      - name: Run tests
+        run: npm run test
+
+      - name: Show Docker logs
+        if: ${{ failure() }}
+        uses: jwalton/gh-docker-logs@v1
+
+  release:
+    name: Release to NPM
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:83]: Release NPM package

File: .github/workflows/test_and_release.yml (hunk 3 above)
Comment (strophy) [reply .github/workflows/test_and_release.yml:16]: This isn't always going to be enough if cache misses, the previous run took 17m

Comment (shumkov) [reply .github/workflows/test_and_release.yml:16]: Oh... You right. Let's keep it.

**Comment by shumkov** [comment]
@strophy please delete `.travis.yml` as well

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**